### PR TITLE
lib: date_time: Check LTE registration status before performing NTP

### PR DESCRIPTION
**Pull request by simensrostad**
Add a guard that prevents the library from performing NTP if the
device is not connected to LTE.

### CODE REVIEW

File: lib/date_time/date_time.c [hunk 1]
```diff
@@ -193,10 +196,41 @@ static int sntp_time_request(struct ntp_servers *server, uint32_t timeout,
...
+}
+#endif /* defined(CONFIG_LTE_LINK_CONTROL) */
+
 static int time_NTP_server_get(void)
 {
 	int err;
 
+#if defined(CONFIG_LTE_LINK_CONTROL)
+	if (!is_connected_to_lte()) {
+		LOG_WRN("Not connected to LTE, skipping NTP UTC time update");
```
Comment (junqingzou) [inline lib/date_time/date_time.c:227]: Can this be changed to LOG_DBG so as to silently skip this round of check?

Comment (simensrostad) [reply lib/date_time/date_time.c]: fixed

**Comment by junqingzou** [comment]
@simensrostad Sorry I did not explain well when I talk to you in Teams. The solution has dependency on LTE_LC module, while for application like SLM you'll get it fail in building. 

I think that you can change the solution to be like, if "date_time: getaddrinfo, error: -11",  skip this round of NTP query.

**Review by rlubos — APPROVED** [review]

**Comment by simensrostad** [comment]
> @simensrostad Sorry I did not explain well when I talk to you in Teams. The solution has dependency on LTE_LC module, while for application like SLM you'll get it fail in building as LTE-LC is not configured at all.
> 
> I think that you can change the solution to be like, if "date_time: getaddrinfo, error: -11", skip this round of NTP query.

Does it really fail during building? I added guards that does not include this new functionality in case `CONFIG_LTE_LINK_CONTROL` is disabled.

**Comment by junqingzou** [comment]
> > @simensrostad Sorry I did not explain well when I talk to you in Teams. The solution has dependency on LTE_LC module, while for application like SLM you'll get it fail in building as LTE-LC is not configured at all.
> > I think that you can change the solution to be like, if "date_time: getaddrinfo, error: -11", skip this round of NTP query.
> 
> Does it really fail during building? I added guards that does not include this new functionality in case `CONFIG_LTE_LINK_CONTROL` is disabled.

Then the original error/warnings will be logged again every one hour.

**Comment by simensrostad** [comment]
> > > @simensrostad Sorry I did not explain well when I talk to you in Teams. The solution has dependency on LTE_LC module, while for application like SLM you'll get it fail in building as LTE-LC is not configured at all.
> > > I think that you can change the solution to be like, if "date_time: getaddrinfo, error: -11", skip this round of NTP query.
> > 
> > 
> > Does it really fail during building? I added guards that does not include this new functionality in case `CONFIG_LTE_LINK_CONTROL` is disabled.
> 
> Then the original error/warnings will be logged again every one hour.

I don't think we should drop subsequent NTP requests if the first one fails. We still want to try other addresses, that is the whole point of having multiple fallback NTP servers.

If you set `DATE_TIME_UPDATE_INTERVAL_SECONDS` to 0, this will disable periodic date time updates. Then you could call `date_time_update_async` in the application when you know that you are connected to LTE.

**Comment by simensrostad** [comment]
Clarified some misunderstandings offline. I still think that this PR is a necessary improvement to the `date_time`library.

**Comment by junqingzou** [comment]
> Clarified some misunderstandings offline. I still think that this PR is a necessary improvement to the `date_time`library.

Thanks for the clarification.

**Review by junqingzou — APPROVED** [review]

**Pull request merged**